pkg/cloud/services/compute/instance_test.go

```diff
@@ -119,73 +119,61 @@ func Test_getPortName(t *testing.T) {
 }
 
 func TestService_getImageID(t *testing.T) {
-	const imageIDA = "ce96e584-7ebc-46d6-9e55-987d72e3806c"
-	const imageIDB = "8f536889-5198-42d7-8314-cb78f4f4755c"
-	const imageIDC = "8f536889-5198-42d7-8314-cb78f4f4755d"
+	const imageID = "ce96e584-7ebc-46d6-9e55-987d72e3806c"
+	const imageName = "test-image"
 
 	tests := []struct {
-		testName  string
-		imageUUID string
-		imageName string
-		expect    func(m *mock.MockImageClientMockRecorder)
-		want      string
-		wantErr   bool
+		testName string
+		image    infrav1.ImageFilter
+		expect   func(m *mock.MockImageClientMockRecorder)
+		want     string
+		wantErr  bool
 	}{
 		{
-			testName:  "Return image uuid if uuid given",
-			imageUUID: imageIDC,
-			want:      imageIDC,
-			expect: func(m *mock.MockImageClientMockRecorder) {
-			},
-			wantErr: false,
-		},
-		{
-			testName:  "Return through uuid if both uuid and name given",
-			imageName: "dummy",
-			imageUUID: imageIDC,
-			expect: func(m *mock.MockImageClientMockRecorder) {
-			},
-			want:    imageIDC,
-			wantErr: false,
+			testName: "Return image ID when ID given",
+			image:    infrav1.ImageFilter{ID: imageID},
+			want:     imageID,
+			expect:   func(m *mock.MockImageClientMockRecorder) {},
+			wantErr:  false,
 		},
 		{
-			testName:  "Return image ID",
-			imageName: "test-image",
+			testName: "Return image ID when name given",
+			image:    infrav1.ImageFilter{Name: imageName},
+			want:     imageID,
 			expect: func(m *mock.MockImageClientMockRecorder) {
-				m.ListImages(images.ListOpts{Name: "test-image"}).Return(
-					[]images.Image{{ID: imageIDA, Name: "test-image"}},
+				m.ListImages(images.ListOpts{Name: imageName}).Return(
+					[]images.Image{{ID: imageID, Name: imageName}},
 					nil)
 			},
-			want:    imageIDA,
 			wantErr: false,
 		},
```

Comment:
Do we want a test case for filtering by tags?

Reply:
I can add, it doesn't hurt